crates/ty_python_semantic/src/types.rs

```diff
@@ -1542,6 +1542,14 @@ impl<'db> Type<'db> {
                 true
             }
 
+            (Type::ClassLiteral(class), Type::SubclassOf(_))
+                if class
+                    .iter_mro(db, None)
+                    .any(|x| ClassBase::Dynamic(DynamicType::Unknown) == x) =>
```

Comment:
I think we should also view classes that inherit from `Any` or `Todo` as assignable to arbitrary `type[]` types

```suggestion
                    .any(|superclass| superclass.is_dynamic()) =>
```

Reply:
I added a `is_dynamic` method to `ClassBase` and replaced this check using it:)